### PR TITLE
Ignore exceptions raised when assembling input kinds.

### PR DESCRIPTION
**Pull request by earthling**
Those inputs will simply be left out of the collection.
We've installed the JMX app and the 'Splunk App for Streams' on our splunk server. Both of these apps have data inputs that cause the code in InputCollection to raise an exception - preventing any inputs from being enumerated. The modification here will just omit troublesome inputs from the InputCollection.

### CODE REVIEW

File: splunk/com/splunk/InputCollection.java [hunk 1]
```diff
@@ -240,35 +240,38 @@ protected static boolean matchesInputName(InputKind kind, String searchFor, Stri
...
+                    // Found an InputKind leaf
+                    InputKind newKind = InputKind.create(relpath);
+                    kinds.add(newKind);
+                } else {
+                    Set<InputKind> subKinds = assembleInputKindSet(thisSubPath);
+                    kinds.addAll(subKinds);
+                }
             }
         }
+        catch (Exception ignored) { }
```
Comment (itay) [inline splunk/com/splunk/InputCollection.java]: Should we add logging here?

**Comment by itay** [comment]
@earthling thanks for the PR! A few notes:

1. Can you sign the contributor agreement: http://www.splunk.com/goto/contributions
2. Can you make the PR target `develop` and not `master`?
3. What was the error that they were throwing? I'm curious if there is a legitimate issue.

File: splunk/com/splunk/InputCollection.java (hunk 1 above)
Comment (earthling) [inline splunk/com/splunk/InputCollection.java]: I would add logging there, but the splunk-sdk doesn't seem to use logging anywhere else. Would java.util.logging be okay?

**Comment by earthling** [comment]
I signed the contributor agreement. I'll try to make the PR target develop, but I'm frankly horrible at git. The stack trace for the exception being swallowed here looks like this:
```

com.splunk.HttpException: HTTP 404 -- <?xml version="1.0" encoding="UTF-8"?>
<response>
  <messages>
    <msg type="ERROR">
 In handler 'jmx': Invalid custom action for this internal handler (handler: jmx, custom action: helloworld, eai action: list).</msg>
  </messages>
</response>

	at com.splunk.HttpException.create(HttpException.java:84)
	at com.splunk.HttpService.send(HttpService.java:386)
	at com.splunk.Service.send(Service.java:1278)
	at com.splunk.HttpService.get(HttpService.java:138)
	at com.splunk.InputCollection.assembleInputKindSet(InputCollection.java:243)
	at com.splunk.InputCollection.assembleInputKindSet(InputCollection.java:268)
	at com.splunk.InputCollection.assembleInputKindSet(InputCollection.java:268)
	at com.splunk.InputCollection.assembleInputKindSet(InputCollection.java:268)
	at com.splunk.InputCollection.refreshInputKinds(InputCollection.java:279)
	at com.splunk.InputCollection.refresh(InputCollection.java:292)
	at com.splunk.InputCollection.refresh(InputCollection.java:26)
	at com.splunk.Resource.validate(Resource.java:174)
	at com.splunk.ResourceCollection.validate(ResourceCollection.java:350)
	at com.splunk.ResourceCollection.values(ResourceCollection.java:357)
```

**Comment by itay** [comment]
@earthling something a bit more odd is going on here - I'm not sure this exception should be thrown.

What's the best way for us to reproduce this?

**Comment by earthling** [comment]
We installed the splunk jmx app (SPLUNK4JMX) and the splunk streams app (splunk_app_stream). Noth of them cause this exception. Our code is just calling into ```Service::getInputs```.

**Comment by itay** [comment]
@earthling coud you rebase the branch against develop so it will use the auto-run tests? If you don't think you can do it, let me know.

**Comment by itay** [comment]
@earthling sorry for the delay. I've run this with both the streamfwd app and SPLUNK4JMX, and it works just fine. So a few more questions:

1. What version of Splunk are you running?
2. What OS are you running?

I just want to reproduce this so we can figure out the right fix.

**Comment by earthling** [comment]
It's likely this was due to some misconfiguration in our environment. I'll try again to reproduce it here. Thanks for getting back to me.

**Comment by shakeelmohamed** [comment]
@earthling we're going to close this due to inactivity. If you're still facing this issue you can open another PR against the `develop` branch.

**Pull request closed without merging**